numba/cuda/compiler.py

```diff
@@ -103,6 +103,33 @@ def run_pass(self, state):
         return True
 
 
+@register_pass(mutates_CFG=False, analysis_only=True)
+class CUDALegalization(AnalysisPass):
+
+    _name = "cuda_legalization"
+
+    def __init__(self):
+        AnalysisPass.__init__(self)
+
+    def run_pass(self, state):
+        # Early return if NVVM 7
+        from numba.cuda.cudadrv.nvvm import NVVM
+        if NVVM().is_nvvm70:
+            return False
+        # NVVM < 7, need to check for charseq
+        typmap = state.typemap
+        for k, v in typmap.items():
+            if isinstance(v, types.Array):
```

Comment:
This may need to also check for record arrays etc?

Reply:
Yes - records can be nested too, so I've reworked it so that the check recurses through fields.

Reply:
Many thanks.